nemo/collections/asr/data_layer.py

```diff
@@ -358,6 +214,321 @@ def data_iterator(self):
+            Defaults to True.
+        trim_silence (bool): Whether to use trim silence from beginning and end of audio signal using
+            librosa.effects.trim().
+            Defaults to False.
+        load_audio (bool): Dataset parameter.
+            Controls whether the dataloader loads the audio signal and transcript or just the transcript.
+            Defaults to True.
+        drop_last (bool): See PyTorch DataLoader.
+            Defaults to False.
+        shuffle (bool): See PyTorch DataLoader.
+            Defaults to True.
+        num_workers (int): See PyTorch DataLoader.
+            Defaults to 0.
+        augmentor (AudioAugmentor or dict): Optional AudioAugmentor or dictionary of str -> kwargs (dict) which is
+            parsed and used to initialize an AudioAugmentor.
+            Note: It is crucial that each individual augmentation has a keyword `prob`, that defines a float
+            probability in the the range [0, 1] of this augmentation being applied. If this keyword is not present,
+            then the augmentation is disabled and a warning is logged.
+    """
+
+    @property
+    @add_port_docs()
+    def output_ports(self):
+        """Returns definitions of module output ports.
+        """
+        return {
+            'audio_signal': NeuralType(
+                ('B', 'T'),
+                AudioSignal(freq=self._sample_rate)
+                if self is not None and self._sample_rate is not None
+                else AudioSignal(),
+            ),
+            'a_sig_length': NeuralType(tuple('B'), LengthsType()),
+            'transcripts': NeuralType(('B', 'T'), LabelsType()),
+            'transcript_length': NeuralType(tuple('B'), LengthsType()),
+        }
+
+    def __init__(
+        self, batch_size=None, pad_id=None, drop_last=False, shuffle=True, num_workers=0,
+    ):
+        super().__init__()
+        self._batch_size = batch_size
+
+        # Set up data loader
+        if self._placement == DeviceType.AllGpu:
+            logging.info("Parallelizing Datalayer.")
+            sampler = torch.utils.data.distributed.DistributedSampler(self._dataset)
+        else:
+            sampler = None
+
+        if batch_size == -1:
+            batch_size = len(self._dataset)
+
+        pad_id = 0 if pad_id is None else pad_id
+        self._dataloader = torch.utils.data.DataLoader(
+            dataset=self._dataset,
+            batch_size=batch_size,
+            collate_fn=partial(seq_collate_fn, token_pad_value=pad_id),
+            drop_last=drop_last,
+            shuffle=shuffle if sampler is None else False,
+            sampler=sampler,
+            num_workers=num_workers,
+        )
+
+    def __len__(self):
+        return len(self._dataset)
+
+    @property
+    def dataset(self):
+        return None
+
+    @property
+    def data_iterator(self):
+        return self._dataloader
+
+
+class AudioToTextCharDataLayer(_AudioToTextDataLayer):
```

Comment:
If `AudioToTextDataLayer` is being deprecated in v0.12, we should probably alias `AudioToTextDataLayer = AudioToTextCharDataLayer` to maintain compat with older training scripts